### DISCUSSION TITLE
Cannot compile the github Repo in Windows using MSYS2 environment, config generates no Makefile

### DISCUSSION THREAD

**Discussion by CyberAkumaZero** [esnet/iperf#1298] [Q&A]
I was having issues using iperf3 in Windows with commands seemingly being unsupported, such as -r. The latest Windows binaries I could find seem to be from 2016 while the github repo is still being updated often. So I figured the only real way to be to get the latest build is to compile it myself. 
 
I admit I an very beginner when it comes to this, but I do have a MSYS2 environment setup that I have compiled other Github projects on. However, when I run ./configure I get no makefile and no error on screen. 

The full output of ./configure is rather large, but here are the last several lines:
 
`checking whether the gcc linker (C:/msys64/mingw64/x86_64-w64-mingw32/bin/ld.exe) supports shared libraries... yes
checking whether -lc should be explicitly linked in... yes
checking dynamic linker characteristics... Win32 ld.exe
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking for gcc... (cached) gcc
checking whether the compiler supports GNU C... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to enable C11 features... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking for ranlib... (cached) ranlib
checking whether ln -s works... no, using cp -pR
checking for library containing floor... none required
checking for library containing socket... no
socket()`

It just ends by saying "socket()", no confirmation or error or makefile being generated. I don't know what I am doing wrong.

**Comment by davidBar-On** [comment]
There is a site that distributes new iperf3 executable for Windows. This is not an iperf3 official site but it seems to be reliable: [https://files.budman.pw/](https://files.budman.pw/) or [https://www.neowin.net/forum/topic/1234695-iperf-3101-windows-build/page/3/](https://www.neowin.net/forum/topic/1234695-iperf-3101-windows-build/page/3/) (in both download is done from the same site).

For building iperf3 under Windows (using [Cygwin](https://www.cygwin.com/)) see: https://www.embeddedsystemtesting.com/2014/08/how-to-compile-iperf3-for-windows.html.

**Reply by NiQ1** [reply]
Is there any way to compile it without Cygwin? I'm trying to build it for arm64, and Cygwin is not available for is and won't be available in the foreseeable future.

**Reply by davidBar-On** [reply]
If [MSYS2](https://www.msys2.org/) is supported on your platform, you can use it instead of Cygwin.  See [MSYS2 iperf3](https://packages.msys2.org/packages/iperf3?variant=x86_64) and this [iperf3 PR](https://github.com/esnet/iperf/pull/1068).

If not, you may try using [iperf2](https://sourceforge.net/projects/iperf2/) which I believe is supported on Windows without using Cygwin.
